getting_started/build-guide/build-with-zephyr.rst

```diff
@@ -0,0 +1,69 @@
+.. _build-with-zephyr:
+
+Build SOF with Zephyr
+#####################
+
+.. contents::
+   :local:
+   :depth: 3
+
+This guide describes the process of building and running |SOF| as a Zephyr
+application.
+
+.. note::
+
+        The example uses ``$ZEPHYR_WORKSPACE`` as the working directory.
+
+Preparation
+***********
+
+The easiest way to build Zephyr is by using its recommended toolchain. Follow
+instructions in the
+`Install a Toolchain <https://docs.zephyrproject.org/latest/getting_started/index.html#install-a-toolchain>`__
+chapter for toolchain installation details.
+
+Check out and build
+*******************
+
+Zephyr uses ``west`` as a source management and building system. Follow the
+Zephyr
+`Getting Started <https://docs.zephyrproject.org/latest/getting_started/index.html#>`__
+guide for dependencies and for ``west`` installation.
+
+Initialize a new ``west`` repository
+
+.. code-block:: bash
+
+    mkdir $ZEPHYR_WORKSPACE
+    cd $ZEPHYR_WORKSPACE
+    west init
+    west update
+
+.. note::
+
+    If you need to checkout a different Zephyr fork or branch you can do something like
+
+.. code-block:: bash
+
+    west init --manifest-url https://github.com/thesofproject/zephyr --manifest-rev intel_adsp_soc_family
+
+This checks out all the Zephyr sources, including |SOF|. After that a firmware
+image can be built and signed:
+
+.. code-block:: bash
+
+    west build -p -d build-apl -b up_squared_adsp zephyr/samples/audio/sof/ -- -DCMAKE_C_FLAGS=-Izephyr/lib/libc/minimal/include/
+    west sign -d build-apl -t rimage -- -k modules/audio/sof/keys/otc_private_key.pem
```

Comment:
I got `FATAL ERROR: rimage not found; either install it or provide --tool-path` here, seem west doesn't help to build the rimage, I think you may also add some documentation here to describe how rimage is built.

Reply:
rimage should be checked out by west ? @lyakh are we missing some init to put rimage in zephyr $PATH.

Reply:
I mean, `west build -p -d build-apl -b up_squared_adsp zephyr/samples/audio/sof/ -- -DCMAKE_C_FLAGS=-Izephyr/lib/libc/minimal/include/` this line doesn't help to build rimage, I think

Reply:
@aiChaoSONG there is a tool called ```west``` that should do this for you and check out rimage and install in your zephyr $PATH

Reply:
@lgirdwood FWIW west isn't even fetching rimage for me, let alone building and installing it.

Reply:
west doesn't support git submodules (which is quite understandable) so this guide is missing at least one `git submodule --init update` command.

Reply:
Support for submodules has just been merged in west. To try it:
```shell
pip3 uninstall west
git clone https://github.com/zephyrproject-rtos/west && \
   (cd west && pip3 install --editable .)
```
Then add `submodules: true` in the SOF section, see incoming documentation at https://github.com/zephyrproject-rtos/zephyr/pull/31241/files"git submodules in projects"